walk/config.go

```diff
@@ -119,22 +133,38 @@ func (*Configurer) KnownDirectives() []string {
 }
 
 func (cr *Configurer) Configure(c *config.Config, rel string, f *rule.File) {
-	wc := getWalkConfig(c)
-	wcCopy := &walkConfig{}
-	*wcCopy = *wc
-	wcCopy.ignore = false
+	if c.Exts[walkNameCached] != nil {
+		// A normal Configurer implementation would process directives and set
+		// c.Exts[walkName] here. However, we've parallelized the tree walk and
+		// processed the configuration ahead of time in configureForWalk. So instead,
+		// the caller of this method (configure) sets c.Exts[walkNameCache] to the
+		// preprocessed configuration. We copy it to c.Exts[walkName] instead of
+		// re-processing directives.
+		c.Exts[walkName] = c.Exts[walkNameCached]
+	} else {
+		// If c.Exts[walkNameCached] was not set, process directives normally.
```

Comment:
Can you explain when it would not be set? I would think it should have always been set during the fs-walk?

Reply:
Updated comment to indicate this branch is only needed for tests. I didn't want to require that all unit tests go through `walk.Walk2` to call `populateCache`. But it makes this code a little awkward.